jetty-infinispan/src/main/java/org/eclipse/jetty/session/infinispan/SessionDataMarshaller.java

```diff
@@ -0,0 +1,119 @@
+    private static final int VERSION = 0;
+
+    @Override
+    public Class<? extends SessionData> getJavaClass()
+    {
+        return SessionData.class;
+    }
+
+    @Override
+    public String getTypeName()
+    {
+        return "org_eclipse_jetty_session_infinispan.SessionData";
+    }
+
+    @Override
+    public SessionData readFrom(ProtoStreamReader in) throws IOException
+    {
+        int version = in.readInt("version");//version of serialized session
+        String id = in.readString("id"); //session id
+        String cpath = in.readString("contextPath"); //context path
+        String vhost = in.readString("vhost"); //first vhost
+
+        long accessed = in.readLong("accessed");//accessTime
+        long lastAccessed = in.readLong("lastAccessed"); //lastAccessTime
+        long created = in.readLong("created"); //time created
+        long cookieSet = in.readLong("cookieSet");//time cookie was set
+        String lastNode = in.readString("lastNode"); //name of last node managing
+
+        long expiry = in.readLong("expiry");
+        long maxInactiveMs = in.readLong("maxInactiveMs");
+
+        SessionData sd = new SessionData(id, cpath, vhost, created, accessed, lastAccessed, maxInactiveMs);
+
+        if (version == 0)
+        {
+            byte[] attributeArray = in.readBytes("attributes");
+            ByteArrayInputStream bais = new ByteArrayInputStream(attributeArray);
+            try (ClassLoadingObjectInputStream ois = new ClassLoadingObjectInputStream(bais))
+            {
+                SessionData.deserializeAttributes(sd, ois);
+            }
+            catch(ClassNotFoundException e)
+            {
+                throw new IOException(e);
+            }
+            sd.setCookieSet(cookieSet);
+            sd.setLastNode(lastNode);
+            sd.setExpiry(expiry);
+            return sd;
+        }
+        else
+            throw new IOException("Unrecognized infinispan session version "+version);
+    }
+
+    @Override
+    public void writeTo(ProtoStreamWriter out, SessionData sdata) throws IOException
+    {
+        out.writeInt("version", VERSION);
+        out.writeString("id", sdata.getId()); //session id
+        out.writeString("contextPath", sdata.getContextPath()); //context path
+        out.writeString("vhost", sdata.getVhost()); //first vhost
+
+        out.writeLong("accessed", sdata.getAccessed());//accessTime
+        out.writeLong("lastAccessed", sdata.getLastAccessed()); //lastAccessTime
+        out.writeLong("created", sdata.getCreated()); //time created
+        out.writeLong("cookieSet", sdata.getCookieSet());//time cookie was set
+        out.writeString("lastNode", sdata.getLastNode()); //name of last node managing
+
+        out.writeLong("expiry", sdata.getExpiry());
+        out.writeLong("maxInactiveMs", sdata.getMaxInactiveMs());
+
+        ByteArrayOutputStream bout = new ByteArrayOutputStream();
+        ObjectOutputStream oout = new ObjectOutputStream(bout);
```

Comment:
`ObjectOutputStream` not closed, use try-with-resources.

Reply:
ok